view/frontend/templates/button.phtml

```diff
@@ -27,7 +27,7 @@ if ($block->shouldDisableBoltCheckout()) { return;
 if ($block->isBoltOnCartDisabled()) { return;
 }
 
-$checkoutButtonUrl = $block->getCheckoutCdnUrl(). '/v1/checkout_button?publishable_key='. $block->getCheckoutKey();
+$checkoutButtonUrl = $block->getCheckoutCdnUrl(). '/v1/checkout_button?publishable_key='. $block->getCheckoutKey().$block->getVariantForInstantButton();
```

Comment:
What about function name? I think instant button is some legacy function we introduce in this file in lines 46-49 and this one is rather for "regular" button.

Reply:
We created the whole $checkoutButonUrl only to be used for Instant button. I can refactor it entirely and just fetch the URL via a function instead?

Reply:
not necessary, I understand the code now